source/_integrations/influxdb.markdown

```diff
@@ -37,41 +39,64 @@ You will still need to create a database named `home_assistant` via InfluxDB's c
 {% configuration %}
 host:
   type: string
-  description: IP address of your database host, e.g., 192.168.1.10
+  description: "**1.xx only** IP address of your database host, e.g., 192.168.1.10"
   required: false
   default: localhost
 port:
   type: integer
-  description: Port to use
+  description: "**1.xx only** Port to use"
   required: false
   default: 8086
 username:
   type: string
-  description: The username of the database user. The user needs read/write privileges on the database
-  required: false
+  description: "**1.xx only** The username of the database user. The user needs read/write privileges on the database."
+  required: inclusive
 password:
   type: string
-  description: The password for the database user account.
-  required: false
+  description: "**1.xx only** The password for the database user account. Needed with `username` configuration variable."
+  required: inclusive
 database:
   type: string
-  description: Name of the database to use. The database must already exist.
+  description: "**1.xx only** Name of the database to use. The database must already exist."
   required: false
   default: home_assistant
 ssl:
   type: boolean
-  description: Use HTTPS instead of HTTP to connect.
+  description: "**1.xx only** Use HTTPS instead of HTTP to connect."
   required: false
   default: false
 verify_ssl:
   type: boolean
-  description: Verify SSL certificate for HTTPS request.
+  description: "**1.xx only** Verify SSL certificate for HTTPS request."
   required: false
   default: true
 path:
   type: string
-  description: Path to use if your InfuxDB is running behind an reverse proxy.
+  description: "**1.xx only** Path to use if your InfuxDB is running behind an reverse proxy."
   required: false
+api_v2:
+  type: boolean
+  description: "**2.xx only** Set to `true` if connecting to a 2.xx installation, do not use otherwise."
+  required: inclusive
+  default: false
```

Comment:
So what if v3 comes out? This feels confusing. I feel that there should be a single option which takes a version as a parameter, defaulting to v1.

Reply:
The thread [here](https://github.com/home-assistant/core/pull/35392#pullrequestreview-421997905) has the reasoning behind this. I agree with the logic and at first I tried to implement it this way but discovered if I switch to a version field that takes a number there's a validation issue I'm not sure how to deal with.  Waiting on a second opinion in the other PR there to see how to best resolve this.

Reply:
@frenck Issue should be resolved now. A solution to the validation issue was found so now both doc and runtime are updated to use an `api_version` config option instead of the v2 specific `api_v2`.